cmd/switch.go

```diff
@@ -36,34 +37,85 @@ func newSwitchCmd(homedir string) *cobra.Command {
 		Example: `# switch the active istioctl version to version=1.7.4, flavor=tetrate and flavor-version=1
 $ getistio switch --version 1.7.4 --flavor tetrate --flavor-version=1`,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			d := &api.IstioDistribution{
-				Version:       flagVersion,
-				Flavor:        flagFlavor,
-				FlavorVersion: int64(flagFlavorVersion),
+			d, err := switchParse(homedir, flagName, flagVersion, flagFlavor, flagFlavorVersion)
+			if err != nil {
+				return err
 			}
-
 			return switchExec(homedir, d)
 		},
 	}
 
 	flags := cmd.Flags()
 	flags.SortFlags = false
-	flags.StringVarP(&flagVersion, "version", "", "", "Version of istioctl e.g. 1.7.4")
+	flags.StringVarP(&flagName, "name", "", "", "Name of istioctl, e.g. 1.9.0-istio-v0")
+	flags.StringVarP(&flagVersion, "version", "", "", "Version of istioctl, e.g. 1.7.4")
 	flags.StringVarP(&flagFlavor, "flavor", "", "", "Flavor of istioctl, e.g. \"tetrate\" or \"tetratefips\" or \"istio\"")
 	flags.IntVarP(&flagFlavorVersion, "flavor-version", "", -1, "Version of the flavor, e.g. 1")
 
-	_ = cmd.MarkFlagRequired("version")
-	_ = cmd.MarkFlagRequired("flavor")
-	_ = cmd.MarkFlagRequired("flavor-version")
-
 	return cmd
 }
 
+// if set name, it should only parse name to distro
```

Comment:
I think we should add this capability into `fetch` command as well. Could you open an issue for tracking that?

Reply:
https://github.com/tetratelabs/getistio/issues/19